internal/reconcilers/status.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/redhat-openshift-ecosystem/operator-certification-operator/internal/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultKubeconfigSecretName        = "kubeconfig"
+	defaultKubeconfigSecretKeyName     = "kubeconfig"
+	defaultGithubApiSecretName         = "github-api-token"
+	defaultGithubApiSecretKeyName      = "GITHUB_TOKEN"
+	defaultPyxisApiSecretName          = "pyxis-api-secret"
+	defaultPyxisApiSecretKeyName       = "pyxis_api_key"
+	defaultDockerRegistrySecretKeyName = ".dockerconfigjson"
+	defaultGithubSSHSecretKeyName      = "id_rsa"
+)
+
+type StatusReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+func NewStatusReconciler(client client.Client, log logr.Logger, scheme *runtime.Scheme) *StatusReconciler {
+	return &StatusReconciler{
+		Client: client,
+		Log:    log,
+		Scheme: scheme,
+	}
+}
+
+func overrideSecretFromSpec(secretDefault, spec string) string {
+	if len(spec) > 0 {
+		return spec
+	}
+	return secretDefault
+}
+
+func (r *StatusReconciler) Reconcile(ctx context.Context, pipeline *certv1alpha1.OperatorPipeline) (bool, error) {
+	origPipeline := pipeline.DeepCopy()
+	pipeline.Status.ObservedGeneration = pipeline.Generation
+	log := r.Log.WithValues("status.observedGeneration", pipeline.Generation)
+
+	kubeconfigSecret := overrideSecretFromSpec(defaultKubeconfigSecretName, pipeline.Spec.KubeconfigSecretName)
+	requeue, err := r.reconcileSecretStatus(ctx, pipeline, "KubeconfigSecret", kubeconfigSecret, defaultKubeconfigSecretKeyName)
+	if requeue || err != nil {
+		return requeue, err
+	}
+
+	githubApiSecret := overrideSecretFromSpec(defaultGithubApiSecretName, pipeline.Spec.GitHubSecretName)
+	requeue, err = r.reconcileSecretStatus(ctx, pipeline, "GithubApiSecret", githubApiSecret, defaultGithubApiSecretKeyName)
+	if requeue || err != nil {
+		return requeue, err
+	}
+
+	pyxisApiSecret := overrideSecretFromSpec(defaultPyxisApiSecretName, pipeline.Spec.PyxisSecretName)
+	requeue, err = r.reconcileSecretStatus(ctx, pipeline, "PyxisApiSecret", pyxisApiSecret, defaultPyxisApiSecretKeyName)
+	if requeue || err != nil {
+		return requeue, err
+	}
+
+	if len(pipeline.Spec.DockerRegistrySecretName) > 0 {
+		requeue, err = r.reconcileSecretStatus(ctx, pipeline, "DockerRegistrySecret", pipeline.Spec.DockerRegistrySecretName, defaultDockerRegistrySecretKeyName)
+		if requeue || err != nil {
+			return requeue, err
+		}
+	}
+
+	if len(pipeline.Spec.GithubSSHSecretName) > 0 {
+		requeue, err = r.reconcileSecretStatus(ctx, pipeline, "GithubSSHSecret", pipeline.Spec.GithubSSHSecretName, defaultGithubSSHSecretKeyName)
+		if requeue || err != nil {
+			return requeue, err
+		}
+	}
```

Comment:
Non Blocking: I know this is the order we were checking before, but maybe it makes sense to move this to after line 99, so this file is easier to read top down.

Reply:
I actually moved it to L71, to put it immediately after the other Github secret.